src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java

```diff
@@ -49,40 +53,50 @@ private FeedbackSessionAttributes(String feedbackSessionName, String courseId) {
         this.isClosingEmailEnabled = true;
         this.isPublishedEmailEnabled = true;
 
+        this.studentDeadlines = new HashMap<>();
+        this.instructorDeadlines = new HashMap<>();
+
         this.timeZone = Const.DEFAULT_TIME_ZONE;
         this.gracePeriod = Duration.ZERO;
 
     }
 
+    private FeedbackSessionAttributes(FeedbackSession feedbackSession) {
```

Comment:
Ok. Can't think of a better way. Though this makes `valueOf` basically redundant. Can resolve it some time in the future.

Reply:
Since I have removed the sub-classes as discussed, `valueOf` is returned to its original state and the constructor is gone. So `valueOf` is no longer redundant; no need to resolve in the future.